docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,192 @@
+# Migration Guide
+
+This document describes how to migrate a project from kubebuilder v1 to v2.
```

Comment:
Lang: this should be self-evident.  If it's not, the title's not doing its job.  You probably want to change the title.

Additional benefit of a more specific title: it allows us to easily add more if we need them in the future.

Reply:
Add a note about how this conversion will end up with a project that looks like a native v2 project -- in some cases it's possible to migrate without as much copy-pasting if you're willing to settle